flink-runtime/src/test/java/org/apache/flink/runtime/jobmanager/TestingJobGraphStoreWatcher.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.jobmanager;
+
+import org.apache.flink.api.common.JobID;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * {@link JobGraphStoreWatcher} implementation for testing purposes.
+ */
+public class TestingJobGraphStoreWatcher implements JobGraphStoreWatcher {
+
+	private JobGraphStore.JobGraphListener jobGraphListener;
+
+	@Override
+	public void start(JobGraphStore.JobGraphListener jobGraphListener) {
+		this.jobGraphListener = jobGraphListener;
+	}
+
+	@Override
+	public void stop() {
+		// noop
+	}
+
+	public void addJobGraph(JobID jobID) {
```

Comment:
Frankly speaking, I do not like package private. It will take some additional burn if we want to do some refactor, especially moving from one package to another.